core/src/main/java/fr/sncf/osrd/new_infra/implementation/reservation/ReservationInfraBuilder.java

```diff
@@ -102,6 +104,21 @@ private ReservationInfra build() {
         return networkBuilder.build();
     }
 
+    /** Returns true if the route is controlled (requires explicit reservation to be used) */
+    private boolean isRouteControlled(ImmutableList<TrackRangeView> trackRanges) {
+        // TODO: eventually, add an optional parameter to RJSRoute
+        for (int i = 1; i < trackRanges.size(); i++) {
+            var prev = trackRanges.get(i - 1).track.getEdge();
+            var next = trackRanges.get(i).track.getEdge();
+            if (prev != next) {
+                if (GraphHelpers.areEdgesLinkedBySwitchBranch(diTrackInfra.getTrackGraph(), prev, next))
```

Comment:
I find it slightly weird to need these helpers, TrackRangeView or similar could embed along a list of TrackEdge, which you could then just query with instanceof